app/views/posts/show.html.erb

```diff
@@ -1,14 +1,28 @@
 <div class="user_c">
-<div class="user_data_post"> 
-  <h2>#<%= @post.title %> by <span><%= @user.name %></span></h2>
-  <h3 class = "post-comment">Comment <%= @post.comments_counter%>,  <span>Likes <%= @post.likes_counter%></span></h3>
-  <p><%= @post.text %></p>  
-</div>
+  <div class="user_data_post"> 
+    <h2>#<%= @post.title %> by <span><%= @user.name %></span></h2>
+    <h3 class = "post-comment">Comments <%= @post.comments_counter%>,  <span>Likes <%= @post.likes_counter%></span></h3>
+    <p><%= @post.text %></p>  
+  </div>
 </div>
 <div>
-<div class="comments">
-<% @comments.each do |comment| %>
-   <h2><%= comment.author.name%>: <%= comment.text%></h2>
-<%end%>
+  <div class="comments">
+  <% @comments.each do |comment| %>
+    <h2><%= comment.author.name%>: <%= comment.text%></h2>
+  <%end%>
+  </div>
+
+  <%=form_with url: user_post_comments_url(@user.id, @post.id), scope: :data do |f| %>
+      <div>
+        <%= f.label :comment %>
+        <%= f.text_area :comment, :required=>true %>
+        <%= f.submit "Comment"%>
+      </div>
+    <% end %>
+     <%=form_with url: user_post_likes_url(@user.id, @post.id), scope: :data do |f| %>
+      <div>
+        <%= f.submit "Like"%>
+      </div>
+    <% end %>
```

Comment:
- [ ] This comment form works fine. However, it is recommended that you create a dedicated view for the comments and serve the form from there. This way, it is easier to scale the app and add any new feature that may be required in the comments. Kindly and a comments view and move this form to `views/comments/new.html.erb`

Reply:
Hi @oloomoses thanks for the review, however, I wanted to ask is this a project requirement because It didn't say in the requirement section.

Reply:
Hi @Wes-Isaac, thanks once again for reaching out for clarification. This has not been mentioned in the requirements, However, It's RoR convention to have separate views for every controller. This way it is easier to scale the app as your project grows bigger and with more functionalities.

Reply:
Hi @oloomoses, thanks for replying, in the previous project it specifically says to add a view for post and user, so if it didn't specify for this project should this change be optional?